Alewis/custom webpack config for sites (#606)
cloudflare/workers-docs: content/tooling/wrangler/webpack.md

```js
module.exports = {

>   target: "webworker",

Reviewer: webworker is the default. Could we remove it?

Author: it feels worth calling out, i think; i'd rather folks not try to overwrite it otherwise, but i should also read the rest of this document to see if we call it out elsewhere.

Author: hmmm. we do, but also all of our examples include it. we should remove it everywhere if we want to be clearer on this.

```js
module.exports = {
  target: "webworker",
  entry: "./index.js",
  mode: "development"
}
```

`webpack.production.js`

```js
module.exports = {
  target: "webworker",
  entry: "./index.js",
  mode: "production"
}
```

### Using with Workers Sites

> Because Wrangler commands are run from the project root, but your site worker is typically nested in a child directory  `workers-site`, you must be mindful when using custom webpack configuration files for Workers Sites projects. We recommend that you keep your webpack config in the workers-site directory, and set `context = __dirname`. This will assure that all of the paths you set in your configuration are evaluated relative to the webpack config, and not the wrangler.toml.

Reviewer 2: IMO this adds more room for confusion than clarity. Also, this situation is not unique to Workers Sites but to all wrangler projects, many of which have multiple build steps not involving wrangler.

Before we merge anything, I would like to discuss if this is the best possible solution. I haven't tested or looked into this, but in my experience, JS developers prefer having all their webpack.config's at root and set entry from there e.g. `entry: "./workers-site/index.js"` . I'd like to get another opinion on this.

I recommend in the meantime moving and shortening this statement to the top of the page:

```suggestion
#Sensible Defaults
 Wrangler commands are run from the project root, so ensure your `entry` and `context` are set appropriately.
```
